api/utils/logging/roundtripper.go

```diff
@@ -0,0 +1,36 @@
+package logging
+
+import (
+	"net/http"
+
+	"github.com/mandelsoft/logging"
+)
+
+func NewRoundTripper(rt http.RoundTripper, logger logging.Logger) *RoundTripper {
+	return &RoundTripper{
+		logger:       logger,
+		RoundTripper: rt,
+	}
+}
+
+// RoundTripper is a http.RoundTripper that logs requests.
+type RoundTripper struct {
+	logger logging.Logger
+	http.RoundTripper
+}
+
+func (t *RoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	// Redact the Authorization header to make sure it doesn't get logged at any point.
+	header := req.Header
+	if key := "Authorization"; req.Header.Get(key) != "" {
+		header = header.Clone()
+		header.Set(key, "***")
```

Comment:
nit: might print first/last two or three characters / `Bearer ` + first few characters (may be helpful in certain cases of debugging).

Reply:
Thought about this but then remembered that basic auth has a different Header and there are cases where basic auth might still be used.

Dont really want to do a string parsing on that header to either get `Basic` or `Bearer` out, so gonna omit for now.